### PR TITLE
[WIP] Build RMM's Python bindings with PTDS

### PR DESCRIPTION
**Pull request by jakirkham**
Defines `CUDA_API_PER_THREAD_DEFAULT_STREAM` for Setuptools `Extensions` built by `setup.py`. This should make sure the per-thread default stream is used by these Python extensions.

~~**Note:** We still need other downstream changes as well ( https://github.com/rapidsai/cudf/issues/5922 ). So we may want to hold on this a bit until those are resolved (hence marked as WIP).~~ Based on discussions with others, it sounds like this change is fine on its own. So marking as ready.

https://docs.nvidia.com/cuda/cuda-runtime-api/stream-sync-behavior.html

cc @kkraus14 @shwina @jrhemstad

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by jakirkham** [comment]
rerun tests

**Comment by kkraus14** [comment]
We should be careful to test at least cudf with this before merging as well as doing some basic benchmarking to see if the extra synchronization caused by mixing PTDS and legacy default streams cause slowdowns.

**Comment by jakirkham** [comment]
cc @pentschev (for vis)

File: python/setup.py [hunk 1]
```diff
@@ -84,6 +84,10 @@ def get_cuda_version_from_header(cuda_include_dir):
 except Exception:
     nthreads = 0
 
+define_macros = [
+    ("CUDA_API_PER_THREAD_DEFAULT_STREAM", None),
```
Comment (harrism) [inline python/setup.py:88]: So does this make this a command line option? How does this work? (Not that familiar with setup.py)

**Comment by jakirkham** [comment]
Sorry was updating this based on feedback from Peter earlier. Have marked it as a draft.

**Comment by harrism** [comment]
@jakirkham can we close this?

**Comment by kkraus14** [comment]
I think we want to keep this open as a point of reference, but no action should be taken on it.

**Comment by jakirkham** [comment]
Ah sorry yeah we can close. I think Peter has PR ( https://github.com/rapidsai/rmm/pull/633 ), which is the successor to this

**Comment by harrism** [comment]
Neither is a great option. Closing this one.

**Comment by jakirkham** [comment]
What do we not like about the other one? Sounds good.

**Comment by harrism** [comment]
See the discussion starting here. https://github.com/rapidsai/rmm/pull/633#issuecomment-757002946

**Pull request closed without merging**